python/cudf/cudf/_lib/pylibcudf/strings/slice.pyx

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+
+from cudf._lib.pylibcudf.column cimport Column
+from cudf._lib.pylibcudf.libcudf.column.column cimport column
+from cudf._lib.pylibcudf.libcudf.scalar.scalar cimport numeric_scalar
+from cudf._lib.pylibcudf.libcudf.scalar.scalar_factories cimport (
+    make_fixed_width_scalar as cpp_make_fixed_width_scalar,
+)
+from cudf._lib.pylibcudf.libcudf.strings cimport substring as cpp_slice
```

Comment:
Not blocking this PR, but wonder if it makes sense to align the libcudf/pylibcudf structure on either `substring` or `slice` rather than using both interchangeably (though don't have much context with strings submodules to know if this was an intentional choice)